### ISSUE TITLE
Angular2 change detection / Tracker.autorun

### ISSUE THREAD

**Issue by stefanholzapfel** [Urigo/angular2-meteor#400]
Hi!

Just stumbled across a problem that seems to be new.

I have a data service where I use observables:

```
this.autorun(() => {
let data = collection.find(publicationName).fetch();
dataStore$.next(data);
}
```

The observable (or BehaviorSubject to be precise) dataStore$ works as expected, but after initial loading, when further changes are triggered via the database from another client, the change arrives in the observable but doesn't trigger Angular change detection and therefore is not displayed.

I can swear the same code was working before the upgrade to Angular2 2.0.1 / Angular2Meteor 0.7.0-beta.3

Any ideas? Thx!

**Comment by Lavan** [comment]
You need to perform the UI update with the correct NgZone. How are you subscribing to the observable?

**Comment by stefanholzapfel** [comment]
The data service (provider) is imported in my core module as a singleton.

It has a method which returns the observable. In the component I just call this method and store the observable. I then render it in the view using the async pipe.

What irritates me is that this same code was working before upgrading to angular2 final. The only thing I changed (beside also upgrading angular-meteor) was that I moved my MongoDB from the localhost to another device.

**Comment by stefanholzapfel** [comment]
Does selecting the browser window also trigger a UI refresh cycle? 
When I use Robomongo to change some data on the server, the view is not updating. But when I reselect the chrome window, it does.

**Comment by stefanholzapfel** [comment]
@barbatus Thanks, but the change in the meteor_reactive.ts doesn't solve the problem for me. Still the observable is fed with the new value, but Zone doesn't refresh until it is triggered by somethin else.

**Comment by barbatus** [comment]
@stefanholzapfel so if you explicitly run callback in the zone like ngZone.run(...), does it help?

**Comment by Lavan** [comment]
I use the `.zone()` operator on the observables that I was wish to update the UI.

**Comment by stefanholzapfel** [comment]
@barbatus Yep refreshing the zone manually works. But that is a bit hacky. Isn't it possible to somehow hook the autorun callback to the zone in meteor-angular directly?

**Comment by barbatus** [comment]
@stefanholzapfel the fix above is about that. When I refactored last time I removed it accidentally (my bad there wan't tests for that), that's why it did work for you before and not now. It's back now, or to be exact, will appear in beta.4 today I guess.

**Comment by stefanholzapfel** [comment]
Ok, I have already tried to manually switch the file with your fix in my node_modules, but that didn't have an effect.

I will try again with the "offical" release in beta 4 and let you now about the outcome.

**Comment by barbatus** [comment]
please try v0.7.0

**Comment by stefanholzapfel** [comment]
Sorry still the same problem, observable change is triggered, but UI doesn't refresh without zone.run().

**Comment by barbatus** [comment]
@stefanholzapfel before publishing I double checked that ngZone runs after autorun callback. It works fine in the demo. So try a simpler case, e.g., add some label that displays collection size, and then see if that label updates with a proper size when the collection changes.

**Comment by stefanholzapfel** [comment]
@barbatus Sorry for my late response! 

Why do we need the zone() operator (if we use MeteorIbservable) when the zone is run anyway after the autorun callback?

Thx in advance!

**Comment by stefanholzapfel** [comment]
I'm asking because I wrap my mongo collections into an observable quite similar as you're doing with MongoObservables and was wondering why my apporach should work without zone.run() if metor-rxjs has to run it? (with .zone())

**Comment by barbatus** [comment]
@stefanholzapfel it's run after MeteorReactive's autorun, not Meteor.autorun.
MeteorObservable.autorun will need zone operator (MeteorObservable.autorun.zone.subscribe) if you want to run the callback in the zone.

**Comment by stefanholzapfel** [comment]
@barbatus Ahhh....That explains it, I'm using Meteor's autorun method and wrap it into an observable manually. 

But still funny that it was working before the update, didn't change anything in my dataservice.

But nevermind, I will just inject the zone in the dataservice and run the zone manually.

**Comment by activeperception** [comment]
Hi @barbatus , how do I get this fix? tried installing 7.0 and 7.0-beta-1 but it does not include the fix.

Thanks
